src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -540,7 +540,10 @@ protected DynamoModel(IStartConfiguration config)
 
             LogWarningMessageEvents.LogWarningMessage += LogWarningMessage;
 
-            StartBackupFilesTimer();
+            // When running test cases, the dispatcher may be null which will cause the timer to
+            // introduce a lot of threads. So the timer will not be started if test cases are running.
+            if (!IsTestMode)
+                StartBackupFilesTimer();
```

Comment:
Hi @Randy-Ma, can we move this check inside `StartBackupFilesTimer` instead? In that case we won't accidentally add another caller which does not have this check in the future. Thanks!